### PR TITLE
FSET-2343: Integrate PHASE2 with the new test provider

### PR DESCRIPTION
**Pull request by fayimora**

### CODE REVIEW

File: app/repositories/onlinetesting/OnlineTestRepository.scala [hunk 1]
```diff
@@ -223,6 +223,15 @@ trait OnlineTestRepository extends RandomSelection with ReactiveRepositoryHelper
     findAndUpdateCubiksTest(cubiksUserId, update, ignoreNotFound = true)
   }
 
+  def updateTestCompletionTime(orderId: String, completedTime: DateTime) = {
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/OnlineTestRepository.scala]: We already have this method - should we not use instead:
  def updateTestCompletionTime2(orderId: String, completedTime: DateTime) = {
    import repositories.BSONDateTimeHandler
    val update = BSONDocument("$set" -> BSONDocument(
      s"testGroups.$phaseName.tests.$$.completedDateTime" -> Some(completedTime)
    ))

    findAndUpdatePsiTest(orderId, update, ignoreNotFound = true)
  }

File: app/repositories/onlinetesting/Phase2TestRepository2.scala [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+trait Phase2TestRepository2 extends OnlineTestRepository with Phase2TestConcern2 {
+  this: ReactiveRepository[_, _] =>
+
+  def getTestGroup(applicationId: String): Future[Option[Phase2TestGroup2]]
+
+  def getTestGroupByUserId(userId: String): Future[Option[Phase2TestGroup2]]
+
+//  def getTestProfileByToken(token: String): Future[Phase2TestGroup]
+
+  def getTestProfileByCubiksId(cubiksUserId: Int): Future[Phase2TestGroupWithAppId]
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/Phase2TestRepository2.scala]: There should be no cubiks specific code in this repo. This is psi only code

File: app/repositories/onlinetesting/Phase2TestRepository2.scala [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait Phase2TestRepository2 extends OnlineTestRepository with Phase2TestConcern2 {
+  this: ReactiveRepository[_, _] =>
+
+  def getTestGroup(applicationId: String): Future[Option[Phase2TestGroup2]]
+
+  def getTestGroupByUserId(userId: String): Future[Option[Phase2TestGroup2]]
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/Phase2TestRepository2.scala]: Is this Cubiks code? Looking for the test group by cubiks user id?

File: app/repositories/onlinetesting/Phase2TestRepository2.scala [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+  def upsertTestGroupEvaluation(applicationId: String, passmarkEvaluation: PassmarkEvaluation): Future[Unit]
+
+  def saveTestGroup(applicationId: String, phase2TestProfile: Phase2TestGroup): Future[Unit]
+
+  def nextTestGroupWithReportReady: Future[Option[Phase2TestGroupWithAppId]]
+
+  def updateGroupExpiryTime(applicationId: String, expirationDate: DateTime): Future[Unit]
+
+  def insertTestResult(appId: String, phase2Test: CubiksTest, testResult: TestResult): Future[Unit]
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/Phase2TestRepository2.scala]: Cubiks specific code. This should be removed

File: app/repositories/onlinetesting/Phase2TestRepository2.scala [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    ))
+  }
+
+  override implicit val bsonHandler: BSONHandler[BSONDocument, Phase2TestGroup2] = Phase2TestGroup2.bsonHandler
+
+  override def getTestGroup(applicationId: String): Future[Option[Phase2TestGroup2]] = {
+    getTestGroup(applicationId, phaseName)
+  }
+
+  override def getTestGroupByUserId(userId: String): Future[Option[Phase2TestGroup2]] = {
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/Phase2TestRepository2.scala]: Is this Cubiks code?

File: app/repositories/onlinetesting/Phase2TestRepository2.scala [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+//  }
+
+  override def nextApplicationsReadyForOnlineTesting(maxBatchSize: Int): Future[List[OnlineTestApplication]] = {
+    val query = inviteToTestBSON(PHASE1_TESTS_PASSED) ++ BSONDocument("applicationRoute" -> BSONDocument("$nin" -> BSONArray("Sdip", "Edip")))
+
+    implicit val reader = bsonReader(repositories.bsonDocToOnlineTestApplication)
+    selectRandom[OnlineTestApplication](query, maxBatchSize)
+  }
+
+  override def getTestProfileByCubiksId(cubiksUserId: Int): Future[Phase2TestGroupWithAppId] = {
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/Phase2TestRepository2.scala]: Cubiks code - we should remove

File: app/repositories/onlinetesting/Phase2TestRepository2.scala [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+  override def saveTestGroup(applicationId: String, phase2TestProfile: Phase2TestGroup) = {
+    val query = BSONDocument("applicationId" -> applicationId)
+    val updateBSON = BSONDocument("$set" -> BSONDocument("testGroups.PHASE2" -> phase2TestProfile))
+
+    val validator = singleUpdateValidator(applicationId, actionDesc = "Saving phase2 test group")
+    collection.update(query, updateBSON).map(validator)
+  }
+
+  override def insertTestResult(appId: String, phase2Test: CubiksTest, testResult: TestResult): Future[Unit] = {
```
Comment (ian-stainer) [inline app/repositories/onlinetesting/Phase2TestRepository2.scala]: Cubiks code - we should remove

File: conf/application.conf [hunk 8]
```diff
@@ -242,13 +230,13 @@ microservice {
 scheduling {
   online-testing {
     send-phase1-invitation-job {
-      enabled = false
```
Comment (ian-stainer) [inline conf/application.conf]: We can't check these in turned on. Will break ATs

File: conf/application.conf [hunk 9]
```diff
@@ -242,13 +230,13 @@ microservice {
...
+      enabled = true
       lockId = "send-phase1-invitation-job-lock-coordinator"
       initialDelaySecs = 30
       intervalSecs = 30
     }
     send-phase2-invitation-job {
-      enabled = false
+      enabled = true
```
Comment (ian-stainer) [inline conf/application.conf]: We can't check these in turned on. Will break ATs

File: conf/application.conf [hunk 10]
```diff
@@ -358,7 +346,7 @@ scheduling {
       intervalSecs = 30
     }
     retrieve-phase1-results-job {
-      enabled = false
+      enabled = true
```
Comment (ian-stainer) [inline conf/application.conf]: We can't check these in turned on. Will break ATs

File: conf/application.conf [hunk 11]
```diff
@@ -372,7 +360,7 @@ scheduling {
       waitSecs = 5
     }
     evaluate-phase1-result-job {
-      enabled = false
+      enabled = true
```
Comment (ian-stainer) [inline conf/application.conf]: We can't check these in turned on. Will break ATs

File: conf/application.conf (hunk 8 above)
Comment (fayimora) [reply conf/application.conf]: Arrgh these were not supposed to be checked in

**Pull request merged**